iceoryx_utils/test/moduletests/test_posix_timer.cpp

```diff
@@ -520,3 +520,24 @@ TIMING_TEST_F(Timer_test, CatchUpPolicySkipToNextBeatCallsLessCallbacksThanASAPT
 
     TIMING_TEST_EXPECT_TRUE(softTimerCounter < asapTimerCounter);
 });
+
+TEST_F(Timer_test, DISABLED_SelfTriggeringTimerWorksAndDoesNotCauseSegFault)
+{
+    Duration selfTriggerTimeout = 1_ns;
+    int repetitions = 100;
+    std::atomic_int counter{0};
+    {
+        Timer sut(selfTriggerTimeout, [&] {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
What is this sleep for? To provoke the error?  If the error depends on the sleep, the test is not well-formed (or the timer used wrongly). I think if the mechanism works properly, there should be no need for a sleep, we should be able to restart it as soon as we are in the callback.

Reply:
I added some documentation